### PR TITLE
test(Badge): migrate BadgeSpec to rtl

### PR DESCRIPTION
**Pull request by marcalexiei**
Part of #6156

### CODE REVIEW

File: test/BadgeSpec.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    const badge = getByTestId('test');
+    badge.tagName.toLocaleLowerCase().should.equal('a');
+    badge.getAttribute('href').should.equal('#');
+  });
+
+  it('Should default to bg="primary"', () => {
+    const { getByTestId } = render(<Badge data-testid="test">Message</Badge>);
+
+    const badge = getByTestId('test');
+    badge.classList.contains('bg-primary');
```
Comment (golota60) [inline test/BadgeSpec.tsx:36]:
```suggestion
    badge.classList.contains('bg-primary').should.be.true;
```

File: test/BadgeSpec.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  it('Should use bg class', () => {
+    const { getByTestId } = render(
+      <Badge bg="danger" data-testid="test">
+        Message
+      </Badge>,
+    );
+
+    const badge = getByTestId('test');
+    badge.classList.contains('bg-primary');
```
Comment (golota60) [inline test/BadgeSpec.tsx:47]:
```suggestion
    badge.classList.contains('bg-danger').should.be.true;
```

File: test/BadgeSpec.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+describe('Badge', () => {
+  it('Should render correctly', () => {
+    const { getByTestId } = render(
+      <Badge bg="primary" pill data-testid="test">
+        Message
+      </Badge>,
+    );
+
+    const badge = getByTestId('test');
+    badge.innerText.should.contains('Message');
```
Comment (golota60) [inline test/BadgeSpec.tsx:14]:
```suggestion
    badge.innerText.should.equal('Message');
```
would be a tad more specific

File: test/BadgeSpec.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  it('should support custom `as`', () => {
+    const { getByTestId } = render(
+      <Badge as="a" href="#" bg="primary" pill data-testid="test">
+        Message
+      </Badge>,
+    );
+
+    const badge = getByTestId('test');
+    badge.tagName.toLocaleLowerCase().should.equal('a');
```
Comment (golota60) [inline test/BadgeSpec.tsx:28]:
```suggestion
    badge.tagName.toLowerCase().should.equal('a');
```

**Review by golota60 — COMMENTED** [review]
looks good, few issues 👍

**Comment by marcalexiei** [comment]
Request changed applied. Sorry for the oversights.

**Review by golota60 — APPROVED** [review]
looks great! 👍  @kyletsang can we merge this?

**Review by kyletsang — APPROVED** [review]

**Comment by kyletsang** [comment]
LGTM.  Thanks!

**Pull request merged**